### PR TITLE
Dependency updates (follow up of #780)

### PR DESCRIPTION
**Pull request by tsmock**
### Description:

Update remaining dependencies

Major updates:
* osmosis 0.44 -> 0.48.3, required function renames from `release` -> `close`, as osmosis updated to Java 8 support
* checkstyle from 8.18 to 9.2.1 -- for this, I had to use the version of `atlas` built in this PR for checkstyle checks due to checkstyle incompatibilities. The checkstyle update also forced some code changes, due to new/fixed checks
* Gradle from 6.9.x to 7.3.3. While this was done independently, the changes are significantly similar to those already present in `atlas-generator`

### Potential Impact:

Downstream users will need to update their dependencies (specifically, `checkstyle`), which will most likely force code changes.

### Unit Test Approach:

N/A

### Test Results:

Timings from `./gradlew run` in `atlas-checks` for NZL:
* 43956s (original)
* 48499s (original)
* 47463s (updates)
* 49334s (updates)

I'm uncertain what happened with the two original runs (10%+ difference), however, the second non-updated run was about the same as the two runs with updates. It could be a simple as different software running on the laptop used.

------

In doubt: [Contributing Guidelines](CONTRIBUTING.md)

### CODE REVIEW

File: dependencies.gradle [hunk 1]
```diff
@@ -50,20 +52,23 @@ project.ext.packages = [
       slf4j: "org.apache.logging.log4j:log4j-slf4j-impl:${versions.log4j}",
     ],
     opencsv: "net.sf.opencsv:opencsv:${versions.opencsv}",
+    google_truth: "com.google.truth:truth:${versions.google_truth}",
```
Comment (matthieun) [inline dependencies.gradle:55]: Nice!

File: gradle/deployment.gradle [hunk 2]
```diff
@@ -23,79 +23,58 @@ signing
     sign configurations.archives
 }
 
-uploadArchives
+publishing
```
Comment (matthieun) [inline gradle/deployment.gradle:26]: So I remember this upload step being very finicky, but that was years back. Have you had any chance to test this new approach?

**Review by matthieun — APPROVED** [review]
This looks good to me, thanks for doing all that maintenance!
Just a question about the upload of archives to maven central...

File: gradle/deployment.gradle (hunk 2 above)
Comment (tsmock) [reply gradle/deployment.gradle:26]: Short answer: no.

Longer answer: the comments indicated that it was due to two problems, https://github.com/travis-ci/travis-ci/issues/9555 and an issue with uploads to the Sonatype repo requiring the same IP. AFAIK, atlas isn't using Travis anymore (but I could definitely be wrong about this), so 9555 isn't an issue. And I don't believe that GitHub actions will rotate IP address per upload.

It _should_ work, but I've never published to Sonatype. I want to say one of the other projects I have worked on did something similar, but I cannot remember which project at this point.

File: dependencies.gradle (hunk 1 above)
Comment (tsmock) [reply dependencies.gradle:55]: You can thank the testing for Checkstyle for that, as I noted in a comment, "Google Truth is needed for checkstyle tests (as of checkstyle 9.2.1)".

**Review by adahn6 — APPROVED** [review]

**Pull request merged**